lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -2659,58 +2670,94 @@ lldb::TypeSystemSP SwiftASTContext::CreateInstance(
     m_description = ss.GetString();
   }
 
-  bool is_repl = extra_options;
   if (is_repl)
     LOG_PRINTF(GetLog(LLDBLog::Types), "REPL detected");
+
+  ModuleSP module_sp = sc.module_sp;
   TargetSP target_sp = typeref_typesystem.GetTargetWP().lock();
-  if (!target_sp)
-    return {};
-  Target &target = *target_sp;
 
   // Make an AST but don't set the triple yet. We need to
   // try and detect if we have a iOS simulator.
-  std::shared_ptr<SwiftASTContextForExpressions> swift_ast_sp(
-      new SwiftASTContextForExpressions(m_description, typeref_typesystem));
+  std::shared_ptr<SwiftASTContext> swift_ast_sp;
+  if (for_expressions) {
+    // Expression context.
+    if (!target_sp && for_expressions) {
+      LOG_PRINTF(GetLog(LLDBLog::Types), "No target for expression typesystem");
+      return {};
+    }
+    swift_ast_sp.reset(new SwiftASTContextForExpressions(
+        m_description,
+        *llvm::cast<TypeSystemSwiftTypeRef>(&typeref_typesystem)));
+    // This is a scratch AST context, mark it as such.
+    swift_ast_sp->m_is_scratch_context = true;
+
+    swift_ast_sp->GetLanguageOptions().EnableCXXInterop =
```

Comment:
Looks like there's some repeated code here, could you extract that to a `SetCompilerOptions` function?

Reply:
I cleaned this up a bit, factoring anything out further into function makes the code less readable.